lib/algae/free.ex

```diff
@@ -111,3 +114,178 @@ defmodule Algae.Free do
     |> wrap()
   end
 end
+
+alias Algae.Free
+alias Algae.Free.{Pure, Roll}
+alias TypeClass.Property.Generator
+alias Witchcraft.{Apply, Chain, Functor, Ord, Setoid}
+import TypeClass
+use Witchcraft
+
+#############
+# Generator #
+#############
+
+defimpl TypeClass.Property.Generator, for: Algae.Free.Pure do
```

Comment:
@icidasset what's the advantage of sticking these all in one file?

Reply:
It's the only way I got it to work 🤷‍♂ 

Otherwise I keep getting weird errors like `TypeClass.Property.Generator.Proto.Algae.Free.Pure` doesn't exist. I thought it might be a `type_class` issue, because all the errors seem to be related to `defimpl` and protocols. But I tried a few different changes in `type_class`, but without any result.

Reply:
Also, to comment on your question in chat.

> Maybe we need to give the compiler hints with requires?

I don't know how a `require` could give the compiler a hint.
Any documentation, or code, I should look at?

Reply:
@expede What was your original reasoning for sticking some of these in one file, whereas others not? (eg. `Maybe` in 1 file and `Reader` multiple files)